src/main/java/io/neonbee/internal/cluster/entity/UnregisterEntityVerticlesHook.java

```diff
@@ -0,0 +1,93 @@
+package io.neonbee.internal.cluster.entity;
+
+import static io.neonbee.hook.HookType.CLUSTER_NODE_ID;
+import static io.vertx.core.Future.succeededFuture;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.neonbee.NeonBee;
+import io.neonbee.hook.Hook;
+import io.neonbee.hook.HookContext;
+import io.neonbee.hook.HookType;
+import io.neonbee.internal.Registry;
+import io.neonbee.internal.cluster.ClusterHelper;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+
+/**
+ * Hooks for unregistering verticle models.
+ */
+public class UnregisterEntityVerticlesHook {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UnregisterEntityVerticlesHook.class);
+
+    /**
+     * This method is called when a NeonBee instance shutdown gracefully.
+     *
+     * @param neonBee     the {@link NeonBee} instance
+     * @param hookContext the {@link HookContext}
+     * @param promise     {@link Promise} to completed the function.
+     */
+    @Hook(HookType.BEFORE_SHUTDOWN)
+    public void unregisterOnShutdown(NeonBee neonBee, HookContext hookContext, Promise<Void> promise) {
+        LOGGER.info("Unregistering models on shutdown");
+        Vertx vertx = neonBee.getVertx();
+        String nodeId = ClusterHelper.getNodeId(neonBee.getVertx());
```

Comment:
```suggestion
        String nodeId = ClusterHelper.getNodeId(vertx);
```

you already have vertx in a local variable (line above)